ComPact/Verifier/PactVerifier.cs

```diff
@@ -53,13 +53,32 @@ public async Task VerifyPactAsync(string path)
 
             if (pact.Interactions != null)
             {
-                if (_config.ProviderBaseUrl == null)
-                {
-                    throw new PactException("Could not verify pacts. Please configure a ProviderBaseUrl.");
+                if (_config.ProviderBaseUrl == null && _config.ProviderHttpClient == null)
+                {
+                    throw new PactException("Could not verify pacts. Please configure a ProviderBaseUrl or a pre-configured ProviderHttpClient.");
                 }
-                using (var client = new HttpClient())
-                {
-                    tests.AddRange(await VerifyInteractions(pact.Interactions, _config.ProviderBaseUrl, client.SendAsync, _config.ProviderStateHandler));
+
+                var existingClient = _config.ProviderHttpClient != null;
```

Comment:
How about something like: var client = _config.ProviderHttpClient ?? new HttpClient();

Reply:
I was originally going to do that, but then I realised that we shouldn't be disposing the client unless we created it, so we either need create _our_ client as a different variable (so it's clear we created it), but that makes the call to use the client more complicated... or we need to do something like this so that we know to only dispose of the client if we were the ones who created it